packages/eslint-plugin-ui5-define-jsdocs/lib/rules/wrong-jsdoc-param-type.js

```diff
@@ -0,0 +1,101 @@
+"use strict";
+
+const _ = require("lodash");
+const helper = require("../utils/helper");
+const messages = require("../messages/manager");
+
+let getParamsWithWrongType = (oJsDocComment, aModuleArguments) => {
+  let aParamsWithWrongType = [];
+  const aParamTags = helper.getJsDocParamTags(oJsDocComment);
+  _.forEach(aParamTags, oParamTag => {
+    const sParamName = _.get(oParamTag, "name");
+    const sActualTypeName = _.get(oParamTag, "type.name");
+    if (!_.isEmpty(sActualTypeName)) {
+      _.forEach(aModuleArguments, oModuleArgument => {
+        const sRightTypeName = helper.createJsDocParamType(
+          oModuleArgument.sModulePath
+        );
+        if (
+          sParamName === oModuleArgument.sArgumentName &&
+          sActualTypeName !== sRightTypeName
+        ) {
+          aParamsWithWrongType.push({
+            oModuleNode: oModuleArgument.oModuleNode,
+            sWrongTypeName: sActualTypeName,
+            sRightTypeName: sRightTypeName,
+            sParamName: sParamName
+          });
+        }
+      });
+    }
+  });
+
+  return aParamsWithWrongType;
+};
+
+let constructRegExp = (sParamType, sArgumentName) => {
+  let sPattern =
```

Comment:
would template strings of ECMA6 be useful in building the regExp?

Reply:
Done